upload/src/addons/SV/WarningImprovements/XF/Service/User/ContentChange.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace SV\WarningImprovements\XF\Service\User;
+
+/**
+ * Class ContentChange
+ *
+ * @package SV\WarningImprovements
+ */
+class ContentChange extends XFCP_ContentChange
+{
+    /**
+     * ContentChange constructor.
+     *
+     * @param \XF\App $app
+     * @param         $originalUserId
+     * @param null    $originalUserName
+     */
+    public function __construct(\XF\App $app, $originalUserId, $originalUserName = null)
+    {
+        parent::__construct($app, $originalUserId, $originalUserName);
+        if ($this->newUserId !== 0) // deleting user
+        {
+            $this->steps[] = 'stepReassignWarningActions';
+        }
+    }
+
+    /**
+     * @throws \XF\PrintableException
+     */
+    protected function stepReassignWarningActions()
+    {
+        /** @var \SV\WarningImprovements\XF\Repository\UserChangeTemp $userChangeTempRepo */
+        $userChangeTempRepo = $this->repository('XF:UserChangeTemp');
+        /** @var \SV\WarningImprovements\XF\Entity\User $targetUser */
+        $targetUser = $this->em()->find('XF:User', $this->newUserId);
```

Comment:
if `$targetUser` is falsey, skip this step

Reply:
That check is already made in `__construct()` unless I'm understanding wrong.